Description: Skip tests which require a certain level of privilege
 During build we cannot bindmount sysfs and cgroupfs in a chroot which leads to
 failures.
Author: Lucas Kanashiro <kanashiro@ubuntu.com>
Forwarded: not-needed
Last-Updated: 2021-05-20

--- a/pkg/cri/server/container_create_linux_test.go
+++ b/pkg/cri/server/container_create_linux_test.go
@@ -189,6 +189,7 @@
 }
 
 func TestContainerCapabilities(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -269,6 +270,7 @@
 }
 
 func TestContainerSpecTty(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -291,6 +293,7 @@
 }
 
 func TestContainerSpecDefaultPath(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -313,6 +316,7 @@
 }
 
 func TestContainerSpecReadonlyRootfs(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -330,6 +334,7 @@
 }
 
 func TestContainerSpecWithExtraMounts(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -391,6 +396,7 @@
 }
 
 func TestContainerAndSandboxPrivileged(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -604,6 +610,7 @@
 }
 
 func TestPrivilegedBindMount(t *testing.T) {
+	t.Skip("It requires privilege to mount sysfs and cgroupfs. Not achievable during the build.")
 	testPid := uint32(1234)
 	c := newTestCRIService()
 	testSandboxID := "sandbox-id"
@@ -754,6 +761,7 @@
 }
 
 func TestPidNamespace(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testPid := uint32(1234)
 	testSandboxID := "sandbox-id"
@@ -795,6 +803,7 @@
 }
 
 func TestNoDefaultRunMount(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testPid := uint32(1234)
 	testSandboxID := "sandbox-id"
@@ -1099,6 +1108,7 @@
 }
 
 func TestMaskedAndReadonlyPaths(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -1187,6 +1197,7 @@
 }
 
 func TestHostname(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testSandboxID := "sandbox-id"
 	testContainerName := "container-name"
@@ -1384,6 +1395,7 @@
 }
 
 func TestPrivilegedDevices(t *testing.T) {
+	t.Skip("It requires privilege to test devices. Not achievable during the build.")
 	testPid := uint32(1234)
 	c := newTestCRIService()
 	testSandboxID := "sandbox-id"
@@ -1446,6 +1458,7 @@
 }
 
 func TestBaseOCISpec(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	c := newTestCRIService()
 	baseLimit := int64(100)
 	c.baseOCISpecs = map[string]*oci.Spec{
--- a/pkg/cri/server/container_create_test.go
+++ b/pkg/cri/server/container_create_test.go
@@ -56,6 +56,7 @@
 const testImageName = "container-image-name"
 
 func TestGeneralContainerSpec(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	testID := "test-id"
 	testPid := uint32(1234)
 	containerConfig, sandboxConfig, imageConfig, specCheck := getCreateContainerTestData()
@@ -69,6 +70,7 @@
 }
 
 func TestPodAnnotationPassthroughContainerSpec(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	if goruntime.GOOS == "darwin" {
 		t.Skip("not implemented on Darwin")
 	}
@@ -277,6 +279,7 @@
 }
 
 func TestContainerAnnotationPassthroughContainerSpec(t *testing.T) {
+	t.Skip("It requires HugeTLB controller enabled which requires mounting cgroupfs. Not achievable during the build.")
 	if goruntime.GOOS == "darwin" {
 		t.Skip("not implemented on Darwin")
 	}
--- a/pkg/cri/server/container_update_resources_linux_test.go
+++ b/pkg/cri/server/container_update_resources_linux_test.go
@@ -29,6 +29,7 @@
 )
 
 func TestUpdateOCILinuxResource(t *testing.T) {
+	t.Skip("It requires some privileges not achievable during the build.")
 	oomscoreadj := new(int)
 	*oomscoreadj = -500
 	for desc, test := range map[string]struct {
